### ISSUE TITLE
Update contrib to d10 compatible releases

### ISSUE THREAD

**Issue by briand44** [uiowa/uiowa#6901]
# Story

As a web team, we need to update contrib to D10 compatible releases so that we can get off of D9 by Nov 1.

# Additional Info

>Updating contrib could be a good next slice, but it will be a biggie. To make it smaller, I separated out core modules that have moved to contrib out as a separate slice. See issue summary: https://github.com/uiowa/uiowa/issues/6856) - @joewhitsitt

Remove unused modules? https://github.com/uiowa/uiowa/issues/5312?

# Proposed Solution

- uiowa_auth https://github.com/uiowa/uiowa/pull/6928
  - [x] assigned
  - [x] done
- drupal/acquia_purge                  1.2.0         requires          drupal/core (^8.9 || ^9)
  - [x] assigned
  - [x] done                                 
- drupal/administerusersbyrole         3.0.0         requires          drupal/core (^8 || ^9)  
  - [x] assigned
  - [x] done                                           
- drupal/better_normalizers            1.0.0-beta5   requires          drupal/core (^8.7.7 || ^9)        
**On first glance we might not need this for d10. Need to confirm that core normalizers for default_content is all we need**    
  - [x] assigned
  - [x] done                       
- drupal/big_pipe_sessionless          2.0.0         requires          drupal/core (^9)        
  - [x] assigned
  - [x] done                                 
- drupal/components                    2.4.0         requires          drupal/core (^8.7.7 || ^9)
**This looks murky, seems like there are some issues with d10 for it.**                
  - [x] assigned
  - [x] done            
- drupal/config_filter                 2.2.0         requires          drupal/core (^8.8 || ^9)        
  - [x] assigned
  - [x] done               
- drupal/config_ignore                 3.0.0-beta1   requires          drupal/core (^8.8 || ^9)    
  - [x] assigned
  - [x] done                                                          
- drupal/crop                          2.2.0         requires          drupal/core (^8.8 || ^9)    
  - [x] assigned
  - [x] done                             
- drupal/cshs                          3.6.0         requires          drupal/core (^8 || ^9)    https://github.com/uiowa/uiowa/pull/6727  
  - [x] assigned
  - [x] done                             
- drupal/devel                         4.1.5         requires          drupal/core (^8.8 || ^9)  #6943 
  - [x] assigned
  - [x] done                          
- drupal/draggableviews                2.0.1         requires          drupal/core (^8 || ^9)    https://github.com/uiowa/uiowa/pull/6729                               
  - [x] assigned
  - [x] done
- drupal/editor_advanced_link          2.0.0         requires          drupal/core (^9.2)     
  - [ ] assigned
  - [ ] done                                 
- drupal/entity_print                  2.6.0         requires          drupal/core (^8 || ^9)       
  - [x] assigned
  - [x] done                            
- drupal/entity_reference_unpublished  1.3.0         requires          drupal/core (^8 || ^9)    https://github.com/uiowa/uiowa/pull/6605                               
  - [x] assigned
  - [x] done
- drupal/externalauth                  1.4.0         requires          drupal/core (^8 || ^9)      
  - [x] assigned
  - [x] done                             
- drupal/field_delimiter               1.1.0         requires          drupal/core (^8 || ^9)  
**waiting on d10 release - https://www.drupal.org/project/field_delimiter/issues/3287503**   
  - [x] assigned
  - [x] done                              
- drupal/focal_point                   1.5.0         requires          drupal/core (^8.8 || ^9)     
  - [ ] assigned
  - [ ] done                            
- drupal/fragments                     2.0.2         requires          drupal/core (^8.7.7 || ^9)  
  - [x] assigned
  - [x] done                             
- drupal/fullcalendar                  1.0.0-alpha3  requires          drupal/core (^8 || ^9)       
**We are on the latest version, there is no d10 compatible version at this time (9/28/23)**
  - [x] assigned
  - [ ] done                            
- drupal/google_analytics              4.0.0         requires          drupal/core (^8.9|^9.0)     
  - [ ] assigned
  - [ ] done                             
- drupal/google_tag                    1.5.0         requires          drupal/core (^8.8 || ^9)     
  - [ ] assigned
  - [ ] done                            
- drupal/hal                           1.0.3         requires          drupal/core (^9)   
**Needs to be done in the same PR as core 10 as it is only d10 compatible and not backwards**         
  - [ ] assigned
  - [ ] done                             
- drupal/heading                       1.6.0         requires          drupal/core (^8 || ^9)   
  - [x] assigned
  - [x] done                                
- drupal/iframe_title_filter           1.1.0         requires          drupal/core (^8 || ^9)     
  - [x] assigned
  - [x] done                              
- drupal/jquery_ui                     1.4.0         requires          drupal/core (^8 || ^9)            
  - [x] assigned
  - [x] done                       
- drupal/jquery_ui_datepicker          1.2.0         requires          drupal/core (^8 || ^9)     
  - [x] assigned
  - [x] done                              
- drupal/jquery_ui_slider              1.1.0         requires          drupal/core (^8 || ^9)     
  - [x] assigned
  - [x] done                              
- drupal/jquery_ui_touch_punch         1.0.0         requires          drupal/core (^8 || ^9) 
  - [x] assigned
  - [x] done                                  
- drupal/layout_builder_restrictions   2.14.0        requires          drupal/core (^8.8.0 || ^9.0)     
  - [x] assigned
  - [ ] done                        
- drupal/layout_builder_shortcuts      1.0.1         requires          drupal/core (^8.8 || ^9)      
  - [x] assigned
  - [x] done                           
- drupal/lazy                          3.11.0        requires          drupal/core (^8.8 || ^9)          
  - [x] assigned
  - [x] done                       
- drupal/mailsystem                    4.3.0         requires          drupal/core (^8.7.7 || ^9)    https://github.com/uiowa/uiowa/pull/6726                       
  - [x] assigned
  - [x] done    
- drupal/media_entity_facebook         3.0.0-beta2   requires          drupal/core (^8 || ^9) 
  - [ ] assigned
  - [ ] done                                  
- drupal/media_entity_file_replace     1.0.0         requires          drupal/core (^8.7.7 || ^9)        
  - [ ] assigned
  - [ ] done                       
- drupal/media_entity_twitter          2.7.0         requires          drupal/core (^8.8 || ^9)           
  - [ ] assigned
  - [ ] done                      
- drupal/media_library_theme_reset     1.1.0         requires          drupal/core (^8 || ^9)      
**This module is deprecated and will be unsupported after November 2023. 
Replacement module has many changes & turns LB tray into a modal, so not switching at this time.**
  - [x] assigned
  - [x] done                             
- drupal/media_thumbnails              1.0.0-alpha6  requires          drupal/core (^8 || ^9)    
  - [ ] assigned
  - [ ] done                               
- drupal/menu_breadcrumb               1.16.0        requires          drupal/core (^8 || ^9)      
  - [x] assigned
  - [x] done                             
- drupal/menu_link_weight              1.0.0-beta5   requires          drupal/core (^8 || ^9)       
  - [x] assigned
  - [x] done                            
- drupal/migrate_plus                  5.2.0         requires          drupal/core (^9.1)       https://github.com/uiowa/uiowa/pull/6602                                
  - [x] assigned
  - [x] done
- drupal/migrate_tools                 5.1.0         requires          drupal/core (^9.1)           
  - [ ] assigned
  - [ ] done                            
- drupal/mimemail                      1.0.0-alpha4  requires          drupal/core (^8.8.2 || ^9)   
  - [ ] assigned
  - [ ] done                            
- drupal/node_view_permissions         1.5.0         requires          drupal/core (^8 || ^9)      
  - [x] assigned
  - [x] done                             
- drupal/oembed_providers              2.0.2         requires          drupal/core (^9.0)            
  - [ ] assigned
  - [ ] done                           
- drupal/override_node_options         2.6.0         requires          drupal/core (^8 || ^9)    
  - [x] assigned
  - [x] done                               
- drupal/paragraphs                    1.14.0        requires          drupal/core (^9.2)         
  - [x] assigned
  - [x] done                              
- drupal/photoswipe                    3.1.1         requires          drupal/core (^9.3)  https://github.com/uiowa/uiowa/pull/6728   
  - [ ] assigned
  - [ ] done                                  
- drupal/private_content               2.0.0-rc3     requires          drupal/core (^8 || ^9)        
  - [x] assigned
  - [x] done                           
- drupal/purge_queuer_coretags         3.2.0         requires          drupal/core (^8.7.7 || ^9)     
  - [x] assigned
  - [x] done                          
- drupal/rabbit_hole                   1.0.0-beta10  requires          drupal/core (^8 || ^9)       
  - [ ] assigned
  - [ ] done                            
- drupal/replicate                     1.0.0         requires          drupal/core (^8 || ^9)       https://github.com/uiowa/uiowa/pull/6600                            
  - [x] assigned
  - [x] done
- drupal/replicate_ui                  1.0.0         requires          drupal/core (^8 || ^9)    
  - [x] assigned
  - [x] done                               
- drupal/reroute_email                 2.1.1         requires          drupal/core (^8.8 || ^9)     
  - [ ] assigned
  - [ ] done                            
- drupal/responsive_table_filter       1.2.0         requires          drupal/core (^8 || ^9)  
  - [x] assigned
  - [x] done                                 
- drupal/robotstxt                     1.4.0         requires          drupal/core (^8.0 || ^9.0)      
  - [x] assigned
  - [x] done                         
- drupal/schema_metatag                2.3.0         requires          drupal/core (^8.0 || ^9.0)    
  - [ ] assigned
  - [ ] done                           
- drupal/simplei                       1.2.0         requires          drupal/core (^8 || ^9)            
  - [ ] assigned
  - [ ] done                       
- drupal/siteimprove                   1.12.0        requires          drupal/core (^8 || ^9)    
  - [ ] assigned
  - [ ] done                               
- drupal/superfish                     1.4.0         requires          drupal/core (^8 || ^9)         
  - [x] assigned
  - [x] done                          
- drupal/swiftmailer                   2.2.0         requires          drupal/core (^8.9.11 || ^9.0.10)     https://github.com/uiowa/uiowa/issues/4623                    
  - [ ] assigned
  - [ ] done
- drupal/theme_permission              2.0.0-beta1   requires          drupal/core (^8 || ^9)    
  - [x] assigned
  - [x] done                               
- drupal/tvi                           1.0.0-rc4     requires          drupal/core (^8 || ^9)   
  - [x] assigned
  - [x] done                                
- drupal/webform                       6.1.3         requires          drupal/core (^8.8 || ^9)     
  - [ ] assigned
  - [ ] done                            
- drupal/workbench_email               2.2.1         requires          drupal/core (^8 || ^9)
  - [x] assigned
  - [x] done

**Comment by briand44** [comment]
Grooming conversation... If we run into contrib updates that are problematic, we can set them aside for a future slice (maybe swiftmailer?).

**Comment by joewhitsitt** [comment]
Trying to correct my oops from earlier https://github.com/uiowa/uiowa/issues/6856#issuecomment-1730060443 

https://github.com/uiowa/uiowa_auth/pull/11

**Comment by joewhitsitt** [comment]
added the list from the previous issue's report as something to start with. linked some existing issues/prs related to a few of these. i will be picking off this list as I can, but more are welcome

**Comment by joewhitsitt** [comment]
removed a few (allowed_formats, core recommended, etc.)

**Comment by joewhitsitt** [comment]
looking into the impact of better_normalizers

**Comment by briand44** [comment]
Planning conversation... Merge and keep plugging away

**Comment by briand44** [comment]
Discussed in planning that we should scan to see if we are clear to remove embed, entity_embed and full calendar and remove as part of this slice.

**Comment by pyrello** [comment]
> drupal/media_library_theme_reset 1.1.0 requires drupal/core (^8 || ^9)
> This module is deprecated and will be unsupported after November 2023.
> Replacement module has many changes & turns LB tray into a modal, so not switching at this time.

Is this captured somewhere as a follow-up?

**Comment by joewhitsitt** [comment]
>Is this captured somewhere as a follow-up?

Yes https://github.com/uiowa/uiowa/issues/6933

**Comment by briand44** [comment]
Fix issue Marie found and then merge and start new PR.

**Comment by pyrello** [comment]
Closing this issue in favor of #6957 to reduce noise.
